src/main/java/frc/robot/filters/DriveFilterManager.java

```diff
@@ -0,0 +1,49 @@
+package frc.robot.filters;
+
+import frc.robot.subsystems.NavXSubSystem;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+/**
+ *
+ * 
+ * @author aheitkamp
+ */
+public class DriveFilterManager {
```

Comment:
consider implementing this class so that it implements the template method pattern.

filters should have a method like:

public DriveInstruction doFilter ( DriveInstruction di,)

then, this class should check enabled, and not even RUN the filter at all if it is false:

public void filter(DriveInstruction originalInstruction){

for (Filter f: filters){
  DriveInstruction curentInstruction = originalInstruction;
  if ( f.enabled() ){
     currentInstruction = f.doFilter(currentInstruction);
}

this will clean up the code a lot, and will ensure that nobody abuses disabled filters to still perform filtering